File: test-fixtures/languages/typescript/micro/user_service.ts
import { verifyToken, runAdminCommand } from "./auth_service";

interface UserInfo {
  id: string | null;
  token: string;
}

export function getUser(token: string): UserInfo {
  const userId = verifyToken(token);
  return { id: userId, token: token };
}

export function updateUser(token: string, action: string): string | null {
  const userId = verifyToken(token);
  if (userId) {
    runAdminCommand(userId, action);
  }
  return userId;
}

File: test-fixtures/languages/kotlin/micro/UserService.kt
package micro

data class UserInfo(val id: String?, val token: String)

class UserService {
    private val authService = AuthService()

    fun getUser(token: String): UserInfo {
        val userId = authService.verifyToken(token)
        return UserInfo(id = userId, token = token)
    }

    fun updateUser(token: String, action: String): String? {
        val userId = authService.verifyToken(token)
        if (userId != null) {
            authService.runAdminCommand(userId, action)
        }
        return userId
    }
}
